js/ajax.php

```diff
@@ -30,94 +30,97 @@
 $return['error']   = false;
 $return['message'] = '';
 
-switch( $_POST['sailthru_action'] )
-{
-
-	case "add_subscriber":
-		$email = trim( $_POST['email'] );
-		if ( ! filter_var( $email, FILTER_VALIDATE_EMAIL ) || empty( $email ) ) {
-			$return['error']   = true;
-			$return['message'] = 'Please enter a valid email address.';
-		} else {
-			$email = filter_var( $email, FILTER_VALIDATE_EMAIL );
-		}
-
-		if ( isset( $_POST['first_name'] ) && !empty( $_POST['first_name'] ) ){
-			$first_name = filter_var(trim($_POST['first_name']), FILTER_SANITIZE_STRING );
-		} else {
-			$first_name = '';
-		}
-
-		if ( isset( $_POST['last_name'] ) && !empty( $_POST['last_name'] ) ){
-			$last_name = filter_var( trim( $_POST['last_name'] ), FILTER_SANITIZE_STRING ) ;
-		} else {
-			$last_name = '';
-		}
-
-		if ( $first_name || $last_name ) {
-
-			$options = array(
-				'vars' => array(
-					'first_name'	=> $first_name,
-					'last_name'		=> $last_name,
-				)
-			);
+if( isset( $_POST['sailthru_action'] ) ) {
+
+	switch( $_POST['sailthru_action'] ) {
 
-		}
+		case "add_subscriber":
+			$email = isset( $_POST['email'] ) ? trim( sanitize_email( $_POST['email'] ) ) : '';
+			if ( ! filter_var( $email, FILTER_VALIDATE_EMAIL ) || empty( $email ) ) {
+				$return['error']   = true;
+				$return['message'] = 'Please enter a valid email address.';
+			} else {
+				$email = filter_var( $email, FILTER_VALIDATE_EMAIL );
+			}
 
-		$subscribe_to_lists = array();
-			if ( !empty($_POST['sailthru_email_list'] ) ) {
+			if ( isset( $_POST['first_name'] ) && !empty( $_POST['first_name'] ) ){
+				$first_name = filter_var( trim( sanitize_text_field( $_POST['first_name'] ) ), FILTER_SANITIZE_STRING );
+			} else {
+				$first_name = '';
+			}
 
-				$lists = explode(',', $_POST['sailthru_email_list']);
+			if ( isset( $_POST['last_name'] ) && !empty( $_POST['last_name'] ) ){
+				$last_name = filter_var( trim( sanitize_text_field( $_POST['last_name'] ) ), FILTER_SANITIZE_STRING ) ;
+			} else {
+				$last_name = '';
+			}
 
-				foreach( $lists as $key => $list ) {
+			if ( $first_name || $last_name ) {
 
-					$subscribe_to_lists[ $list ] = 1;
+				$options = array(
+					'vars' => array(
```

Comment:
suggestion: Use new php array syntax here if the Wordpress plugin can support it.